Mathlib/Data/Bitvec/Core.lean

```diff
@@ -310,48 +311,77 @@ attribute [local simp] Nat.zero_add Nat.add_zero Nat.one_mul Nat.mul_one Nat.zer
+  simp [bitsToNat_toList]; clear P
+  unfold bitsToNat List.foldl
   -- generalize the accumulator of foldl
   generalize h : 0 = x;
-  conv in add_lsb x b => rw [← h]; clear h
+  conv in addLsb x b =>
+    rw [← h];
   simp
+  clear h
   induction' xs with x xs generalizing x
   · simp
-    unfold List.foldl add_lsb
+    unfold addLsb
+    simp
     simp [Nat.mul_succ]
   · simp
-    apply xs_ih
+    rename_i b' _
+    have s₁: addLsb (List.foldl addLsb (addLsb x b') xs) b =
+      List.foldl addLsb (addLsb x b') xs + List.foldl addLsb (addLsb x b') xs + cond b 1 0 := by
+        rfl
+    have _ :  addLsb 0 b = 0 + 0 + cond b 1 0  := by
+      rfl
+    have s₃ : 0 + 0 + cond b 1 0 = cond b 1 0 := by simp
+    have s₄ : addLsb 0 b = cond b 1 0 := by
+      apply s₃
+    have s₅ : ∀ n : Nat, n + n = n * 2 := by
+      intro n
+      conv =>
+        lhs
+        rw[←Nat.mul_one n,←Nat.mul_add]
+    have s₆: (List.foldl addLsb (addLsb x b') xs)
+      + (List.foldl addLsb (addLsb x b') xs) + cond b 1 0 =
+     (List.foldl addLsb (addLsb x b') xs) * 2 + cond b 1 0 := by
+      rw [s₅ _]
+    have s₇ : addLsb (List.foldl addLsb (addLsb x b') xs) b
+      = (List.foldl addLsb (addLsb x b') xs) * 2 + cond b 1 0 := by
+      rw[s₁, s₆]
+    have s₈: addLsb (List.foldl addLsb (addLsb x b') xs) b =
+     (List.foldl addLsb (addLsb x b') xs) * 2 + addLsb 0 b := by
+     rw[s₇,s₄]
+    rw [Nat.add_comm]
+    exact s₈
 #align bitvec.to_nat_append Bitvec.toNat_append
 
-theorem bits_toNat_decide (n : ℕ) : Bitvec.toNat (decide (n % 2 = 1) ::ᵥ nil) = n % 2 :=
-  by
-  simp [bits_to_nat_to_list]
-  unfold bits_to_nat add_lsb List.foldl cond
-  simp [cond_to_bool_mod_two]
+-- Porting Note: the mathlib3port version of the proof was :
+--  simp [bits_to_nat_to_list]
+--  unfold bits_to_nat add_lsb List.foldl cond
+--  simp [cond_to_bool_mod_two]
+theorem bits_toNat_decide (n : ℕ) : Bitvec.toNat (decide (n % 2 = 1) ::ᵥ nil) = n % 2 := by
+  simp [bitsToNat_toList]
+  unfold bitsToNat addLsb List.foldl
+  simp [Nat.cond_decide_mod_two, -Bool.cond_decide]
 #align bitvec.bits_to_nat_to_bool Bitvec.bits_toNat_decide
 
 theorem ofNat_succ {k n : ℕ} :
     Bitvec.ofNat (succ k) n = Bitvec.ofNat k (n / 2)++ₜdecide (n % 2 = 1) ::ᵥ nil :=
   rfl
 #align bitvec.of_nat_succ Bitvec.ofNat_succ
 
-theorem toNat_ofNat {k n : ℕ} : Bitvec.toNat (Bitvec.ofNat k n) = n % 2 ^ k :=
-  by
+theorem toNat_ofNat {k n : ℕ} : Bitvec.toNat (Bitvec.ofNat k n) = n % 2 ^ k := by
   induction' k with k ih generalizing n
   · simp [Nat.mod_one]
     rfl
-  · rw [of_nat_succ, to_nat_append, ih, bits_to_nat_to_bool, mod_pow_succ, Nat.mul_comm]
+  · rw [ofNat_succ, toNat_append, ih, bits_toNat_decide, mod_pow_succ, Nat.mul_comm]
 #align bitvec.to_nat_of_nat Bitvec.toNat_ofNat
 
 /-- Return the integer encoded by the input bitvector -/
 protected def toInt : ∀ {n : Nat}, Bitvec n → Int
   | 0, _ => 0
-  | succ n, v =>
-    cond (head v) (Int.negSucc <| Bitvec.toNat <| Not <| tail v)
```

Comment:
is there an incorrect align for `bnot`?

Reply:
The [docs](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/Data/Bool/Basic.html) suggest that `bnot` became `not` in lean4/mathlib4

Reply:
Indeed; I'm curious why mathport output `Not` here

Reply:
Not sure. But I see no align in the [source](https://github.com/leanprover/lean4/blob/5349a089e5bb6905eaca1a09e2b587488f8e4715/src/Init/Prelude.lean#L979-L981). Is this some default behaviour when no align is found or is there a possibility that the aligns are elsewhere?

Reply:
The align is [here](https://github.com/leanprover-community/mathlib4/blob/8280a0e1ed38b070f3ab155921a90490e6a71e00/Mathlib/Init/Data/Bool/Basic.lean#L19)